File: 프로그래머스/1/12928. 약수의 합/약수의 합.py
def solution(n):
    end = int(n**0.5)
    answer = 0
    
    for i in range(1, end+1):
        if n % i == 0:
            print(i, n//i)
            answer += i
            if i**2 != n:
                answer += n // i
                
    return answer

File: 프로그래머스/1/12928. 약수의 합/약수의 합.swift
func solution(_ n:Int) -> Int {
    return getDivisor(n).reduce(0, +)
}

func getDivisor(_ n: Int) -> [Int] {
    guard n != 0 else { return [0] }
    
    var divisor = [Int]()
    
    for i in 1...n {
        if n % i == 0 {
            divisor.append(i)
        }
    }
    
    return divisor
}